components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -814,8 +863,8 @@ Field AtmosphereOutput::get_field(const std::string& name, const std::shared_ptr
   } else if (m_diagnostics.find(name) != m_diagnostics.end() && field_mgr==m_sim_field_mgr) {
     const auto& diag = m_diagnostics.at(name);
     return diag->get_diagnostic();
-  } else if (m_fields_alt_name.find(name) != m_fields_alt_name.end() && field_mgr->has_field(m_fields_alt_name.at(name))) {
-    return field_mgr->get_field(m_fields_alt_name.at(name));
+  } else if (m_fields_alt_name.find(name) != m_fields_alt_name.end()) {
+    return get_field(m_fields_alt_name.at(name),field_mgr);
   } else {
     EKAT_ERROR_MSG ("Field " + name + " not found in output field manager or diagnostics list, or requesting a diag not on the simualation field manager.");
```

Comment:
Suggestion: given that we have _several_ FM's in this class, we should print _which_ FM we were querying, cause I suspect it might help debugging what stage of the execution went wrong. We could pass the string with the FM type rather than the FM itself, and retrieve the FM inside the fcn, so that we can print the FM type in the message.